webpack/assets/javascripts/react_app/components/ConfirmModal/ConfirmModal.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import { Modal, Button, ModalVariant } from '@patternfly/react-core';
+import { translate as __ } from '../../common/I18n';
+import { noop } from '../../common/helpers';
+import { closeConfirmModal } from './ConfirmModalActions';
+import {
+  selectConfirmCB,
+  selectIsOpen,
+  selectLabel,
+  selectMessage,
+  selectModalProps,
+} from './ConfirmModalSelectors';
+
+export const ConfirmModal = props => {
+  const [isLoading, setLoading] = React.useState(false);
+  const isOpen = useSelector(selectIsOpen);
+  const label = useSelector(selectLabel);
+  const message = useSelector(selectMessage);
+  const onConfirm = useSelector(selectConfirmCB);
+  const modalProps = useSelector(selectModalProps);
+
+  const dispatch = useDispatch();
+  const closeModal = () => dispatch(closeConfirmModal());
+  const handleConfirm = () => {
+    setLoading(true);
+    onConfirm(closeModal);
```

Comment:
why is the closeModal passed as argument?

Reply:
I was thinking to let the consumer decide whether to close it based on an API response for example,
but I still want to play with that and see if we have a strong use case for that

Reply:
IMHO if you do not have use-case in mind, it's better not to do it, simplier code always wins for me.
If you need it in the future, you can always add that :)

Reply:
right, I agree with that!